backend/apps/dashboard/widgets.py

```
@@ -0,0 +1,93 @@
+    id: str
+    title: str
+    description: str
+    default_width: int = 4
+    default_height: int = 3
+    icon: str = '📊'
+    permissions: List[str] = None
+
+class WidgetRegistry:
+    """Registry for dashboard widgets."""
+    
+    _widgets = {}
```

Ruff (0.15.21)

[warning] Mutable default value for class attribute

(RUF012)